packages/repository-json-schema/src/build-schema.ts

```diff
@@ -122,19 +105,20 @@ export function metaToJsonProperty(meta: PropertyDefinition): JSONSchema {
     result = propDef;
   }
 
-  propertyType = stringTypeToWrapper(propertyType);
+  const wrappedType = stringTypeToWrapper(propertyType);
+  const resolvedType = resolveType(wrappedType);
 
-  if (isComplexType(propertyType)) {
-    Object.assign(propDef, {$ref: `#/definitions/${propertyType.name}`});
-  } else if (propertyType === Date) {
+  if (resolvedType === Date) {
     Object.assign(propDef, {
       type: 'string',
       format: 'date-time',
     });
-  } else {
+  } else if (isBuiltinType(resolvedType)) {
     Object.assign(propDef, {
-      type: <JSONSchemaTypeName>propertyType.name.toLowerCase(),
+      type: resolvedType.name.toLowerCase(),
     });
+  } else {
+    Object.assign(propDef, {$ref: `#/definitions/${resolvedType.name}`});
   }
```

Comment:
I re-organized the branches in a different order, that's why the diff looks bigger than the changes actually are.